applications/launchpad/gui-react/src/components/Inputs/AmountInput/index.tsx

```diff
@@ -0,0 +1,135 @@
+import { ChangeEvent, useState } from 'react'
+import { useDispatch } from 'react-redux'
+
+import SvgQuestion from '../../../styles/Icons/Question'
+import Button from '../../Button'
+import Text from '../../Text'
+
+import t from '../../../locales'
+import MessagesConfig from '../../../config/helpMessagesConfig'
+import { tbotactions } from '../../../store/tbot'
+
+import {
+  IconWrapper,
+  InputContainer,
+  InputWrapper,
+  StyledAmountInput,
+  StyledInput,
+  Currency,
+  TransactionFee,
+  ErrorContainer,
+} from './styles'
+import { AmountInputProps } from './types'
+import { useTheme } from 'styled-components'
+
+const whatIsDecimalSeparator = () => {
+  const n = 1.1
+  return n.toLocaleString().substring(1, 2)
+}
+
+const ds = whatIsDecimalSeparator()
+
+const AmountInput = ({
+  value = 0,
+  disabled,
+  onChange,
+  icon,
+  error,
+  withError,
+  maxDecimals,
+  currency,
+  autoFocus,
+  withFee,
+  fee,
+  testId,
+}: AmountInputProps) => {
+  const dispatch = useDispatch()
+  const theme = useTheme()
+
+  const [valStr, setValStr] = useState(value.toString())
+
+  const onChangeLocal = (e: ChangeEvent<HTMLInputElement>) => {
+    // Remove all non-digits and delimiter characters:
```

Comment:
this smells, hopefully doesn't break too soon xD

Reply:
It was bit challenging, but should work fine. It just removes anything else than numbers and delimiter. And delimiter is detected by the locale.